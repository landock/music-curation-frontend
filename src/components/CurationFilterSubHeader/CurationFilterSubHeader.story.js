import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import CurationFilterSubHeader from '.';
import { tags } from '../../fixtures/test_data';

storiesOf('CurationFilterSubHeader', module).add('default', () => (
  <CurationFilterSubHeader tags={tags} />
));
