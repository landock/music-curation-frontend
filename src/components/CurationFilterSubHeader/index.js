import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import TagCollection from '../TagCollection';
import { Form, Grid } from 'semantic-ui-react';

class CurationFilterSubHeader extends PureComponent {
  render() {
    const statusOptions = [
      { value: 'draft', text: 'Draft' },
      { value: 'published', text: 'Published' },
    ];
    const environmentOptions = [
      { value: 'production', text: 'Production' },
      { value: 'development', text: 'Development' },
    ];

    return (
      <Form>
        <Grid columns={3} style={{ background: 'black', marginBottom: '20px' }}>
          <Grid.Row textAlign="left">
            <Grid.Column textAlign="right">
              <span style={{ color: 'white' }}>Filters:</span>
            </Grid.Column>
            <Grid.Column>
              <Form.Field>
                <label style={{ color: 'white' }}>Environment</label>
                <Form.Select options={environmentOptions} />
              </Form.Field>
              <Form.Field>
                <label style={{ color: 'white' }}>Status</label>
                <Form.Select options={statusOptions} />
              </Form.Field>
            </Grid.Column>
            <Grid.Column>
              <p>
                <strong style={{ color: 'white' }}>Tags</strong>
              </p>
              <TagCollection tags={this.props.tags} />
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Form>
    );
  }
}

CurationFilterSubHeader.propTypes = {
  environments: PropTypes.arrayOf(PropTypes.string),
  statuses: PropTypes.arrayOf(PropTypes.string),
  tags: PropTypes.arrayOf(PropTypes.object),
};

export default CurationFilterSubHeader;
