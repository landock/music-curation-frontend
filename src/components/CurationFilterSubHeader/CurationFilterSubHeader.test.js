import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import CurationFilterSubHeader from '.';

it('CurationFilterSubHeader: default', () => {
  const component = renderer.create(<CurationFilterSubHeader />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
