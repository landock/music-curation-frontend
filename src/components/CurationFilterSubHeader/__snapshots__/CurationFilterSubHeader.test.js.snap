// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurationFilterSubHeader: default 1`] = `
<form
  action={undefined}
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="ui three column grid"
    style={
      Object {
        "background": "black",
        "marginBottom": "20px",
      }
    }
  >
    <div
      className="left aligned row"
    >
      <div
        className="right aligned column"
      >
        <span
          style={
            Object {
              "color": "white",
            }
          }
        >
          Filters:
        </span>
      </div>
      <div
        className="column"
      >
        <div
          className="field"
        >
          <label
            style={
              Object {
                "color": "white",
              }
            }
          >
            Environment
          </label>
          <div
            className="field"
          >
            <div
              aria-busy={undefined}
              aria-disabled={undefined}
              aria-expanded={false}
              aria-multiselectable={undefined}
              className="ui selection dropdown"
              content={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              required={undefined}
              role="listbox"
              tabIndex={0}
              type={undefined}
            >
              <div
                aria-live="polite"
                className="text"
                role="alert"
              />
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                className="menu transition"
              >
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={true}
                  className="selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Production
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Development
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="field"
        >
          <label
            style={
              Object {
                "color": "white",
              }
            }
          >
            Status
          </label>
          <div
            className="field"
          >
            <div
              aria-busy={undefined}
              aria-disabled={undefined}
              aria-expanded={false}
              aria-multiselectable={undefined}
              className="ui selection dropdown"
              content={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              required={undefined}
              role="listbox"
              tabIndex={0}
              type={undefined}
            >
              <div
                aria-live="polite"
                className="text"
                role="alert"
              />
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                className="menu transition"
              >
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={true}
                  className="selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Draft
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Published
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="column"
      >
        <p>
          <strong
            style={
              Object {
                "color": "white",
              }
            }
          >
            Tags
          </strong>
        </p>
        <span>
          No tags
        </span>
      </div>
    </div>
  </div>
</form>
`;
